### PR TITLE
ckan 1.16.1 -> 1.29.2

### PR DESCRIPTION
**Pull request by ymarkus**
Please review this for MacOS, I'm not sure if my change breaks that.

###### Motivation for this change
CKAN is heavily out-of-date

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/ckan/default.nix [hunk 1]
```diff
@@ -1,44 +1,34 @@
...
   pname = "ckan";
-  version = "1.16.1";
+  version = "1.28.0";
 
-  src = fetchFromGitHub {
-    owner = "KSP-CKAN";
-    repo = "CKAN";
-    rev = "v${version}";
-    sha256 = "0lfvl8w09lakz35szp5grfvhq8xx486f5igvj1m6azsql4n929lg";
+  src = fetchurl {
```
Comment (danieldk) [inline pkgs/games/ckan/default.nix:7]: This switches from building from source to downloading a precompiled binary. Building from source is highly prefered in nixpkgs. Did you bump into issues building from source?

**Review by danieldk — CHANGES_REQUESTED** [review]

File: pkgs/games/ckan/default.nix (hunk 1 above)
Comment (ymarkus) [reply pkgs/games/ckan/default.nix:7]: Sorry, should have given a bit more information.
Yes, I tried a lot to build it from source, but for some reason I keep getting this error:
```
MSBuild auto-detection: using msbuild version '' from '/nix/store/vq32vzsz9v7swaf8jlh0sdvkfkfk5042-msbuild-16.3+xamarinxplat.2019.07.26.14.57/bin'.
Error parsing solution file at [...]/src/CKAN/CKAN.sln: Invalid Image
An error occurred when executing task 'Restore-Nuget'.

[...]

Error: One or more errors occurred. (NuGet: Process returned an error (exit code 1).)
	NuGet: Process returned an error (exit code 1).
```

Seems like [this issue on the CKAN repo](https://github.com/KSP-CKAN/CKAN/issues/2842) might be related, but that was solved a few months ago by simply adding msbuild as a dependency, which doesn't work for me.

I've tried mono5, mono6 (which is 6.0.0 in nixos-stable, maybe this is an issue?), dotnetPackages.NUnitConsole and msbuild, as stated [in the build requirements](https://github.com/KSP-CKAN/CKAN/wiki/Build-Environment) and followed [the build guide](https://github.com/KSP-CKAN/CKAN/wiki/Building-CKAN). In the end, I gave up and reverted to simply use the provided .exe, as that is what is being built anyway.

**Comment by lordcirth** [comment]
Thanks for doing this! A while ago I spent a few hours trying to get their new build system to work and gave up.

**Comment by lordcirth** [comment]
Can this please be merged? I've been running this out of a manual nix-build from the branch.

**Comment by danieldk** [comment]
I don't know enough about .NET packaging or mono. Perhaps @jonringer ?

File: pkgs/games/ckan/default.nix [hunk 2]
```diff
@@ -1,44 +1,34 @@
...
+      --add-flags $out/bin/ckan.exe \
+      --set LD_LIBRARY_PATH $libraries
   '';
 
   meta = {
     description = "Mod manager for Kerbal Space Program";
     homepage = "https://github.com/KSP-CKAN/CKAN";
     license = stdenv.lib.licenses.mit;
-    maintainers = [ stdenv.lib.maintainers.Baughn ];
+    maintainers = [ stdenv.lib.maintainers.Baughn stdenv.lib.maintainers.ymarkus ];
```
Comment (SuperSandro2000) [inline pkgs/games/ckan/default.nix:31]:
```suggestion
  meta = with stdenv.lib; {
...
    license = licenses.mit;
    maintainers = with maintainers; [ Baughn ymarkus ];
```

**Comment by ymarkus** [comment]
I had some time to re-review my own request. As of CKAN 1.29.2, it uses [Cake](https://cakebuild.net/) to build. As I can't find anything about this in nixpkgs or the documentation, I guess it has to be set up first. Not really worth the hassle if you ask me considering there is a fine EXE in the releases that definitely works fine. 
How shall we continue?

**Comment by SuperSandro2000** [comment]
> I had some time to re-review my own request. As of CKAN 1.29.2, it uses [Cake](https://cakebuild.net/) to build. As I can't find anything about this in nixpkgs or the documentation, I guess it has to be set up first. Not really worth the hassle if you ask me considering there is a fine EXE in the releases that definitely works fine.
> How shall we continue?

we can use that and patch it if you prefer. The C# support in nix is not that great at this point tbh.

**Comment by ymarkus** [comment]
Ok, I've pushed a new commit. We're now going straight to ckan 1.29.2 and using the provided exe. Please check if it was done properly.

Also, @Baughn please check if you still want to be a maintainer for this package or if I shall remove you from the list.

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

**Comment by jonringer** [comment]
> I don't know enough about .NET packaging or mono. Perhaps @jonringer ?

We don't really have any Cake support, patching the distributables is fine for now.

**Pull request merged**